shopinvader_anonymous_partner/models/res_partner.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2023 ACSONE SA/NV
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import secrets
+import typing
+
+from odoo import _, api, fields, models
+
+COOKIE_NAME = "shopinvader-anonymous-partner"
+COOKIE_MAX_AGE = 86400 * 365
+
+
+class Response(typing.Protocol):
+    def set_cookie(
+        self,
+        key: str,
+        value: str,
+        max_age: int,
+        secure: bool,
+        httponly: bool,
+        samesite: typing.Literal["lax", "strict", "none"],
+    ) -> None:
+        ...
+
+
+class Cookies(typing.Protocol):
+    def get(self, key: str) -> typing.Optional[str]:
+        ...
+
+
+class ResPartner(models.Model):
+
+    _inherit = "res.partner"
+
+    anonymous_token = fields.Char()
+
+    _sql_constraints = [
+        (
+            "anonymous_token_unique",
+            "EXCLUDE (anonymous_token WITH =) WHERE (anonymous_token IS NOT NULL)",
+            "This token is already used!",
+        )
+    ]
+
+    @api.model
+    def _create_anonymous_partner__token(self):
```

Comment:
what is the meaning of the double underscore ?

Reply:
can we have the response in the args  in order to set some additionnal values like the (shopinvader) backend asked in the request ?

Reply:
> what is the meaning of the double underscore ?

That's because with a single underscore it would give the impression to create a token, while here it creates the partner and returns the token. The other creates the partner and sets the cookie. Well... I may be overthinking :)

Reply:
> can we have the response in the args in order to set some additionnal values like the (shopinvader) backend asked in the request ?

I don't understand your question. Can you elaborate / give an example?

Reply:
Use case:  you have multiple web sites, you want to know on which website the user has registred. 

If you want to implement it; you can extract some info from the request and write it on the user.